sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -220,34 +220,41 @@ object NestedColumnAliasing {
   /**
    * Returns two types of expressions:
    * - Root references that are individually accessed
-   * - [[GetStructField]] or [[GetArrayStructFields]] on top of other [[ExtractValue]]s
-   *   or special expressions.
+   * - GetStructField*(_: AttributeReference). GetStructField* means nested GetStructField.
    */
-  private def collectRootReferenceAndExtractValue(e: Expression): Seq[Expression] = e match {
-    case _: AttributeReference => Seq(e)
-    case GetStructField(_: ExtractValue | _: AttributeReference, _, _) => Seq(e)
-    case GetArrayStructFields(_: MapValues |
-                              _: MapKeys |
-                              _: ExtractValue |
-                              _: AttributeReference, _, _, _, _) => Seq(e)
-    case es if es.children.nonEmpty => es.children.flatMap(collectRootReferenceAndExtractValue)
-    case _ => Seq.empty
+  private def collectRootReferenceAndGetStructField(e: Expression): Seq[Expression] = {
```

Comment:
This function is only key change. Looking at the change of this function and the added test should explain what I want to do pretty clearly.